contracts/fungible-token.aes

```diff
@@ -57,16 +63,20 @@ contract FungibleToken =
   public function allowance_from(addr: address) : int =
     Map.lookup_default((addr, Call.caller), state.allowances, 0)
 
+  // TODO check negative allowances possible
   public stateful function transfer_from(from: address, to: address, value: int) =
     require(allowance_from(from) >= value, "The provided value is larger than what is allowances.")
+    // extract calculations froms state updates
```

Comment:
Not sure what you mean by that.

The idea is to protect from reentrancy here and that's why we first update the state, then transferring the value.

Reply:
I want to separate math and state operations